app/controllers/api/v2/meursing_measures_controller.rb

```diff
@@ -0,0 +1,78 @@
+module Api
+  module V2
+    class MeursingMeasuresController < ApiController
+      DEFAULT_INCLUDES = %w[
+        additional_code
+        geographical_area
+        measure_components
+        measure_components.duty_expression
+        measure_type
+      ].freeze
+
+      def index
+        render json: serialized_meursing_measures
+      end
+
+      private
+
+      def serialized_meursing_measures
+        Api::V2::Measures::MeursingMeasureSerializer.new(presented_meursing_measures, serializer_options).serializable_hash
+      end
+
+      def serializer_options
+        {
+          include: DEFAULT_INCLUDES,
+          meta: { resolved_duty_expression: resolved_duty_expression },
+        }
+      end
+
+      def resolved_duty_expression
+        resolved_measure_components.map(&:formatted_duty_expression).join(' ')
+      end
+
+      def resolved_measure_components
+        MeursingMeasureComponentResolverService.new(root_measure, meursing_measures).call
+      end
+
+      def presented_meursing_measures
+        meursing_measures.map { |measure| Api::V2::Measures::MeursingMeasurePresenter.new(measure) }
+      end
+
+      def meursing_measures
+        @meursing_measures ||= root_measure
+          .meursing_measures_for(additional_code_id)
+          .actual
+          .eager(
+            :additional_code,
+            :geographical_area,
+            :measure_components,
+            :measure_type,
+            measure_components: [:duty_expression],
+          )
+          .all
+          .select(&:current?)
```

Comment:
Does this need `current?` - shouldn't the `.actual` filter by timestamp? If not, ideally it would be better to add the filtering to the db query rather than filtering the returned dataset?

Reply:
Spoke about this offline. So I've inherited a broken TimeMachine mechanism from the Measure model. This wasn't noticed previously but was noticed for MeursingMeasure rows since most rows inherit their validity_end_date from the some generating regulation. The business logic for this is complex, needs fixing globally and wouldn't add an unnoticeable benefit performance wise for the complexity of writing the explicit query to do the logic and joins in the database.

I'm going to leave this until we approach the fix for the measures model